format-as-currency.js

```diff
@@ -65,7 +65,9 @@ angular
       })
 
       ngModel.$parsers.push(function (value) {
-
+        // ignore non-numeric characters
+        value = value.replace(/[a-zA-Z!\?>:;\|<@#%\^&\*\)\(\+\/\\={}\[\]_]/g, '')
```

Comment:
That would be nice but, after the first invocation of this parser function, 'value' is the fully formatted string with the currency symbol and possibly '.' or ',' e.g. '$5,678.00'.